medusa/backup_node.py

```diff
@@ -393,6 +415,21 @@ def backup_snapshots(storage, manifest, node_backup, node_backup_cache, snapshot
         raise e
 
 
+def check_missing_files(all_files_in_storage, manifest_objects, srcs):
+    # contents of the storage are described as ManifestObjects, so we need to take just the paths
+    manifest_paths = map(lambda o: o.path, manifest_objects)
+    missing_in_storage = list()
+    # iterate through manifest paths and their corresponding local file names
+    for manifest_path, src in zip(manifest_paths, srcs):
```

Comment:
**Thought:** This can get a little expensive if we have a lot of manifests.
Instead of checking if the files we want to upload are present in the data directory, we're cycling through all past manifests to list files that would be missing.
The design I had in mind was to stop checking the past manifests at all, and replace our [_cached_objects](https://github.com/thelastpickle/cassandra-medusa/blob/09874d7df38d3461dd863e8a4cd6ff203997c91d/medusa/backup_node.py#L46) variable with the list of existing files in the data directory. The rest of the code would be left untouched then, for the most part.
wdyt?

Reply:
I have to admit I have not fully understood how `replace_or_remove_if_cached` actually caches things.

I just worked with the understanding that it takes the list of files to back up (`srcs`) and splits them into two groups 
- the ones that `needs_backup`, which is just a list of strings, just like `srcs`. These are the paths on the local nodes.
- the ones that are `already_backed_up`, which is a list of `ManifestObject`. These are the paths in the storage, together with size, digest and a timestamp.

I've added a third group, with is a list of local paths, so strings like `srcs`, but their order in the list corresponds to the `already_backed_up`. I've done this because to re-upload them, I'd have to somehow convert the path in the storage back to the path on the node, which I couldn't easily figure out.

When we `check_missing_files` we give it the last two groups - a list of ManifestObjects and their corresponding local paths - and a dict of files in the storage keyed by keyspace and table (into a set for quick lookup).

So we are dealing with
- all files in the data folder for the given node
- manifest objects for a given keyspace and table.

We don't deal with any manifests as such, we only `_make_manifest_object()` in the `replace_or_remove_if_cached`. 

I would really like to avoid changing the caching code. However, I'm open to the idea of dropping it because it simplifies things.